### PR TITLE
Retry restore

### PR DESCRIPTION
**Pull request by KevinRansom**
Retry the restore

### CODE REVIEW

**Comment by forki** [comment]
Cool

File: fcs/build.fsx [hunk 1]
```diff
@@ -77,11 +77,12 @@ Target "Restore" (fun _ ->
     // We assume a paket restore has already been run
     runDotnet __SOURCE_DIRECTORY__ "restore FSharp.Compiler.Service.sln -v n"
     for p in [ "../packages.config" ] do
-        ExecProcess (fun info ->
-            info.FileName <- FullName @"./../.nuget/NuGet.exe"
-            info.WorkingDirectory <- FullName @"./.."
-            info.Arguments <- sprintf "restore %s -PackagesDirectory \"%s\" -ConfigFile \"%s\""   (FullName p) (FullName "./../packages") (FullName "./../NuGet.Config")) TimeSpan.MaxValue
-        |> assertExitCodeZero           
+        let executeProcess () =
```
Comment (forki) [inline fcs/build.fsx]: I think you should add a parameter "trials" and only retry 5 times

Comment (KevinRansom) [reply fcs/build.fsx]: I intended it to only retry 1 time.  I figured that would be sufficient to eliminate the annoyance.  And ... how is your vacation?

**Comment by forki** [comment]
From all we learned in paket and FAKE over the years - make it retry
multiple times. ;-)



Kevin Ransom (msft) <notifications@github.com> schrieb am Mi., 13. Juni
2018, 07:36:

**Comment by matthid** [comment]
@forki You mean retry until the stack overflows? ;)

**Comment by forki** [comment]
Dude!

**Comment by KevinRansom** [comment]
@forki, you guys are too funny :-)

**Comment by forki** [comment]
https://github.com/fsprojects/Paket/pull/3251
This is what @matthid is referring to. A bug in paket's retry logic that let stack overflow at -84 trials. Reported yesterday - fixed by same (C#) person today.

**Comment by KevinRansom** [comment]
Let's see how this goes, if the failure still happens an annoying amount, we can retry multiple times, and perhaps try a delay and stuff.

**Comment by KevinRansom** [comment]
@forki,that is too funny.  Software is hard ....

**Comment by dsyme** [comment]
I agree that retrying 5 times would be good, may as well do it now rather than later

File: fcs/build.fsx [hunk 2]
```diff
@@ -77,11 +77,15 @@ Target "Restore" (fun _ ->
...
-            info.FileName <- FullName @"./../.nuget/NuGet.exe"
-            info.WorkingDirectory <- FullName @"./.."
-            info.Arguments <- sprintf "restore %s -PackagesDirectory \"%s\" -ConfigFile \"%s\""   (FullName p) (FullName "./../packages") (FullName "./../NuGet.Config")) TimeSpan.MaxValue
-        |> assertExitCodeZero           
+        let rec executeProcess count =
+            let result = ExecProcess (fun info ->
+                info.FileName <- FullName @"./../.nuget/NuGet.exe"
+                info.WorkingDirectory <- FullName @"./.."
+                info.Arguments <- sprintf "restore %s -PackagesDirectory \"%s\" -ConfigFile \"%s\""   (FullName p) (FullName "./../packages") (FullName "./../NuGet.Config")) TimeSpan.MaxValue
+            if result <> 0 && count > 1 then  executeProcess (count - 1) else 0
```
Comment (brettfo) [inline fcs/build.fsx]: This should read `... else result` so that if the 5 tries are exhausted the last error code is returned.

**Comment by KevinRansom** [comment]
@brettfo Oh yeah ….

**Comment by forki** [comment]
Yay!

Kevin Ransom (msft) <notifications@github.com> schrieb am Do., 14. Juni
2018, 05:51:

**Pull request merged**